forc-test/src/lib.rs

```diff
@@ -98,6 +107,26 @@ impl TestResult {
             }
         }
     }
+
+    /// Return `TestDetails` from the span of the function declaring this test.
+    pub fn details(&self) -> anyhow::Result<TestDetails> {
+        let file_path = self
+            .span
+            .path()
+            .ok_or_else(|| anyhow::anyhow!("Missing span for test function"))?
+            .to_owned();
+        let span_start = self.span.start();
+        let file_str = fs::read_to_string(&*file_path)?;
+        let line_number = file_str[..span_start]
+            .chars()
+            .into_iter()
+            .filter(|&c| c == '\n')
+            .count();
```

Comment:
I would love for spans to have the line number info, as provided by the parser, but if this is working and accurate then it's good enough. :+1: